test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -748,25 +748,38 @@ private static void deleteAllNodeShutdownMetadata() throws IOException {
+            ".ds-ilm-history-*"
+        };
+
         boolean includeHidden = minimumNodeVersion().onOrAfter(Version.V_7_7_0);
         try {
-            //remove all indices except ilm history which can pop up after deleting all data streams but shouldn't interfere
-            final Request deleteRequest = new Request("DELETE", "*,-.ds-ilm-history-*");
+            final Request deleteRequest = new Request("DELETE", "*,-" + String.join(",-", indicesToIgnore));
             deleteRequest.addParameter("expand_wildcards", "open,closed" + (includeHidden ? ",hidden" : ""));
-            RequestOptions allowSystemIndexAccessWarningOptions = RequestOptions.DEFAULT.toBuilder()
-                .setWarningsHandler(warnings -> {
-                    if (warnings.size() == 0) {
-                        return false;
-                    } else if (warnings.size() > 1) {
-                        return true;
-                    }
-                    // We don't know exactly which indices we're cleaning up in advance, so just accept all system index access warnings.
-                    final String warning = warnings.get(0);
-                    final boolean isSystemIndexWarning = warning.contains("this request accesses system indices")
-                        && warning.contains("but in a future major version, direct access to system indices will be prevented by default");
-                    return isSystemIndexWarning == false;
-                }).build();
-            deleteRequest.setOptions(allowSystemIndexAccessWarningOptions);
+
+            if (minimumNodeVersion().onOrAfter(Version.CURRENT)) {
+                // feature reset deletes system indices
+                final Request postRequest = new Request("POST", "/_features/_reset");
+                adminClient().performRequest(postRequest);
+            } else {
+                RequestOptions allowSystemIndexAccessWarningOptions = RequestOptions.DEFAULT.toBuilder()
+                    .setWarningsHandler(warnings -> {
+                        if (warnings.size() == 0) {
+                            return false;
+                        } else if (warnings.size() > 1) {
+                            return true;
+                        }
+                        // We don't know exactly which indices we're cleaning up in advance,
+                        // so just accept all system index access warnings.
+                        final String warning = warnings.get(0);
+                        final boolean isSystemIndexWarning = warning.contains("this request accesses system indices") &&
+                            warning.contains("but in a future major version, direct access to system indices will be prevented by default");
+                        return isSystemIndexWarning == false;
+                    }).build();
+                deleteRequest.setOptions(allowSystemIndexAccessWarningOptions);
```

Comment:
Should we still swallow system index access warnings, given we're using the reset API prior to this request so there shouldn't be any system indices left to delete?

I think it's fine if you left this here on purpose to prevent surprise errors for now, but we should open an issue to clean it up later if that's the case.

Reply:
This ought to be the code path where we have old nodes in the cluster that don't have a feature reset API. It will probably help to put in a specific version number and comment explaining that this only works when all of your nodes have the ability to reset features.